module/VuFind/src/VuFind/Search/UrlQueryHelper.php

```diff
@@ -311,32 +311,34 @@ public function addFacet($field, $value, $operator = 'AND')
     {
         // Facets are just a special case of filters:
         $prefix = ($operator == 'NOT') ? '-' : ($operator == 'OR' ? '~' : '');
-        return $this->addFilter($prefix . $field . ':"' . $value . '"');
-    }
 
-    /**
-     * Add a filter to the parameters.
-     *
-     * @param string $filter Filter to add
-     *
-     * @return UrlQueryHelper
-     */
-    public function addFilter($filter)
-    {
         $params = $this->urlParams;
 
         // Add the filter:
         if (!isset($params['filter'])) {
             $params['filter'] = [];
         }
-        $params['filter'][] = $filter;
+        $params['filter'][] = $prefix . $field . ':"' . $value . '"';
 
         // Clear page:
         unset($params['page']);
 
         return new static($params, $this->queryObject, $this->config, false);
     }
 
+    /**
+     * Add a filter to the parameters.
+     *
+     * @param string $filter Filter to add
+     *
+     * @return UrlQueryHelper
+     */
+    public function addFilter($filter)
+    {
+        [$field, $value] = $this->parseFilter($filter);
```

Comment:
Have you tested whether this works as expected with complicated multi-part checkbox filters (e.g. `format:Book OR title:test`)? If not, maybe we should add a test case for that scenario....

Reply:
Oh no. Of course it doesn't, but with the normalization in JS we shouldn't need this change, so I'll just revert it.

Reply:
Maybe I misunderstood. If the complex filters are a result of CustomFilters, then the JS side is unaffected.

Reply:
As discussed on Slack, I was confusing hidden filters and checkbox filters, so I think all is well. Closing out this thread now.